src/android/src/main/java/io/radar/cordova/RadarCordovaPlugin.java

```diff
@@ -461,21 +471,90 @@ public void startTrip(final JSONArray args, final CallbackContext callbackContex
-        Radar.startTrip(options);
+        Radar.startTrip(options,new Radar.RadarTripCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus radarStatus) {
+                if (radarStatus == Radar.RadarStatus.SUCCESS) {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK));
+                } else {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.ERROR,radarStatus.toString()));
+                }
+            }
+        });
 
         callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK));
     }
 
-    public void completeTrip(final JSONArray args, final CallbackContext callbackContext) throws JSONException {
-        Radar.completeTrip();
+    public void getTripOptions(final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+      RadarTripOptions tripOptions = Radar.getTripOptions();
+      JSONObject tripOptionsObj = new JSONObject();
+      if (tripOptions != null) {
+        tripOptionsObj = tripOptions.toJson(); 
+      }
+      PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, tripOptionsObj);
+      callbackContext.sendPluginResult(pluginResult);
+    }
 
-        callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK));
+    public void updateTrip(final JSONArray args, final CallbackContext callbackContext) throws JSONException {
+        final JSONObject optionsObj = args.getJSONObject(0);
+        JSONObject tripOptionsObj = optionsObj.getJSONObject("options");
+
+        RadarTripOptions options = RadarTripOptions.fromJson(tripOptionsObj);
+        RadarTrip.RadarTripStatus status = RadarTrip.RadarTripStatus.UNKNOWN;
+
+        if (optionsObj.has("status")) {
+          String statusStr = optionsObj.getString("status");
+          if (statusStr != null) {
+            if (statusStr.equalsIgnoreCase("started")) {
+              status = RadarTrip.RadarTripStatus.STARTED;
+            } else if (statusStr.equalsIgnoreCase("approaching")) {
+              status = RadarTrip.RadarTripStatus.APPROACHING;
+            } else if (statusStr.equalsIgnoreCase("arrived")) {
+              status = RadarTrip.RadarTripStatus.ARRIVED;
+            } else if (statusStr.equalsIgnoreCase("completed")) {
+              status = RadarTrip.RadarTripStatus.COMPLETED;
+            } else if (statusStr.equalsIgnoreCase("canceled")) {
+              status = RadarTrip.RadarTripStatus.CANCELED;
+            }
+          }
+        }
+
+        Radar.updateTrip(options, status, new Radar.RadarTripCallback() {
+            @Override
+            public void onComplete(@NonNull Radar.RadarStatus radarStatus) {
+                if (radarStatus == Radar.RadarStatus.SUCCESS) {
+                    callbackContext.sendPluginResult(new PluginResult(PluginResult.Status.OK,radarStatus.toString()));
```

Comment:
@brettguenther @kochis Nit: Add a space between arguments

(lots of these, applies everywhere)